bot/exts/halloween/spooknamerate.py

```diff
@@ -0,0 +1,377 @@
+        self.name = f"{random.choice(self.first_names)} {random.choice(self.last_names)}"
+        await self.data.set('name', self.name)
+
+        await channel.send("Let's move on to the next name!\nAnd the next name is...\n"
+                           f"**{self.name}**!\nTry to spookify that... :smirk:")
+
+        self.poll = False  # accepting responses
+
+    @announce_name.before_loop
+    async def wait_till_scheduled_time(self) -> None:
+        """Waits till the next day's 12AM if crossed it, otherwise waits till the same day's 12AM."""
+        now = datetime.utcnow()
+        if now.hour < 12:
+            twelve_pm = now.replace(hour=12, minute=0, seconds=0, microsecond=0)
+            time_left = twelve_pm - now
+            await asyncio.sleep(time_left.seconds)
+            return
+
+        tomorrow_12pm = now + timedelta(days=1)
+        tomorrow_12pm = tomorrow_12pm.replace(hour=12, minute=0, second=0, microsecond=0)
+        await asyncio.sleep((tomorrow_12pm - now).seconds)
+
+    async def get_responses_list(self, final: bool = False) -> Embed:
+        """Returns an embed containing the responses of the people."""
+        channel = await self.get_channel()
+
+        embed = Embed(
+            color=Colours.red,
+            title="",
+            description=""
+        )
+
+        if await self.messages.items():
+            if final:
+                embed.title = "Spook Name Rate is about to end!"
+                embed.description = ("This Spook Name Rate round is about to end in 2 hours! You can review "
+                                     "the entries below! Have you rated other's names?")
+            else:
+                embed.title = "All the spookified names!"
+                embed.description = "See a list of all the entries entered by everyone!"
+        else:
+            embed.title = "No one has added an entry yet..."
+
+        for message_id, data in await self.messages.items():
+
+            data = json.loads(data)
+
+            embed.add_field(
+                name=(await self.bot.fetch_user(data['author'])).name,
+                value=f"[{(data)['name']}](https://discord.com/channels/{Client.guild}/{channel.id}/{message_id})",
+            )
+
+        return embed
+
+    async def get_channel(self) -> Union[TextChannel, None]:
+        """Gets the seasonalbot-channel after waiting until ready."""
+        await self.bot.wait_until_ready()
+        channel = self.bot.get_channel(self.channel_id)
+        if not channel:
+            logger.warning("Channel is None!")
+        return channel
+
+    @staticmethod
+    def load_json(file: Path) -> Dict[str, str]:
+        """Loads a JSON file and returns its contents."""
+        with file.open('r', encoding='utf-8') as f:
+            return json.load(f)
+
+    @staticmethod
+    def in_allowed_month() -> bool:
+        """Returns whether running in the limited month."""
+        return datetime.utcnow().month == MONTH_TO_LIMIT_TO
```

Comment:
We need to allow month override for development. Use `Client.month_override` from constant and when this is `None`, only then use `utcnow`.